src/erasure-code/jerasure/ErasureCodeJerasure.cc

```diff
@@ -240,7 +240,7 @@ int ErasureCodeJerasureReedSolomonRAID6::parse(ErasureCodeProfile &profile,
 {
   int err = ErasureCodeJerasure::parse(profile, ss);
   profile.erase("m");
-  m = 2;
+  err |= to_int("m", profile, &m, DEFAULT_M, ss);
```

Comment:
i don't really in favor of this. two reasons:

1. we are overwriting the profile provided by user without warnings.
1. `ErasureCodeJerasure::parse()` has already parse and set `m` for us.

instead, i think a better fix is to
```c++
int err = ErasureCodeJerasure::parse(profile, ss);
if (m != DEFAULT_M) {
  *ss << "ReedSolomonRAID6: m=" << m << " must be 2 for RAID6: revert to 2\n";
}
```